### ISSUE TITLE
"Favorite" icon action for a lesson is not accessible

### ISSUE THREAD

**Issue by brianhogg** [gocodebox/lifterlms#2852]
### Reproduction Steps

1. Using a keyboard, attempt to access and click on the "favorite" action for a lesson

### Expected Behavior

I'm able to navigate to the favorite element, and mark a lesson as a favorite.

### Actual Behavior

Because the favorite action is just an icon, and it's contained within a link, it's not accessible.

This issue has be recreated:
+ [ ] Locally
+ [ ] On a staging site
+ [ ] On a production website
+ [ ] With only LifterLMS and a default theme

**Comment by brianhogg** [comment]
We'll want to break this out of the `a` for the lesson and make it a button, using `aria-pressed` true/false to indicate it's toggled or not.

**Comment by brianhogg** [comment]
![image](https://github.com/user-attachments/assets/b5fb2c6b-fb40-4fb3-8d41-825d68322d14)

Something like this, where we add a new row for the favorite action + the has quiz/assignment. The favorite action is a button outside of the link.
